### ISSUE TITLE
queue files not written when using action() syntax (and omrelp)

### ISSUE THREAD

**Issue by mbiebl** [rsyslog/rsyslog#96]
Version: 8.2.2 (and 7.6.3)

rsyslog does not create any queue files when using the new style action syntax.

To test this, I first created an old-style syntax config like the following.

```
module(load="imuxsock") 
module(load="omrelp")
$WorkDirectory /var/spool/rsyslog

$ActionQueueType LinkedList
$ActionQueueFileName test.legacy
$ActionResumeRetryCount -1
$ActionQueueSaveOnShutdown on
$ActionQueueDiscardMark 90000
$ActionQueueSize 100000
*.* :omrelp:localhost:20514;RSYSLOG_TraditionalFileFormat
```

Then I ran rsyslogd -d -n -f test.conf and created a few messages via "logger".
I don't have any server running on localhost, so I expect rsyslog to spool them messages to disk on shutdown
If I then terminate rsyslog via CTRL+C, rsyslog creates the queue files
```
sudo ls -l /var/spool/rsyslog/
insgesamt 12
-rw------- 1 root root 6391 Jun 18 17:03 test.legacy.00000001
-rw------- 1 root root  500 Jun 18 17:03 test.legacy.qi
```

I did the same test using the new style syntax:
```
module(load="imuxsock") 
module(load="omrelp")
$WorkDirectory /var/spool/rsyslog

*.* action(type="omrelp" target="localhost" port="20514" template="RSYSLOG_TraditionalFileFormat" queue.type="LinkedList" queue.size="100000" queue.filename="test.new" queue.saveonshutdown="on" queue.discardmark="90000" queue.highwatermark="1" queue.syncqueuefiles="on")
```

Here, rsyslog does not create any spool files (when shutting down).

I've tested both version 8.2.2 and 7.6.3

**Comment by Smana** [comment]
Hi,

Same problem for me.
OS : debian 7
rsyslog version : 


Legacy syntax is ok
```
module(load="omrelp")
$ActionQueueType LinkedList
$ActionQueueFileName analytics-spool
$ActionResumeRetryCount -1
$ActionQueueSaveOnShutdown on
$ActionQueueHighWaterMark 10000
if $fromhost == "10.115.77.6" then :omrelp:10.115.77.8:20514
```

Syntax with action() doesn't create the spool files
```
if $fromhost == '10.115.77.6' then {
       action(name="analytics"
             type="omrelp"
             target="10.115.77.8"
             port="20514"
             queue.fileName="analytics-spool"
             queue.highwatermark="10000"
             queue.saveonshutdown="on"
             #action.resumeretrycount="-1"
             queue.type="LinkedList")
}
```

Let me know how to give you more details
Regards

**Comment by rgerhards** [comment]
Can you create a startup debug log for both cases. I want to see how the internal queue structures get populated.

**Comment by Smana** [comment]
Please find the debug files here : 
https://gist.github.com/470d5a83e70ffd3f29ed.git

**Comment by Smana** [comment]
@rgerhards 
Are the logs what you expected ? Do you have enough information ?

Regards

**Comment by Smana** [comment]
Hello, 

Any updates ? :)

**Comment by Smana** [comment]
Hello,

I still have the same issue.
In order to spool properly to disk, i used the workaround with the legacy syntaxe.
```
ruleset(name="forward"){
    $ActionName somequeue
    $ActionQueueType LinkedList
    $ActionQueueFileName somequeue-spool
    $ActionResumeRetryCount -1
    $ActionQueueSaveOnShutdown on
    $ActionQueueHighWaterMark 100000
    $ActionSendTCPRebindInterval 1000
    if $fromhost-ip == '10.18.113.9' then :omrelp:remoteserver:20514
}
```

But now i'm facing a performance issue which seems to be solved by increasing the "windowSize" as follows.
```
ruleset(name="forward"){
  if $programname startswith 'mytag.' then {
    action(name="somequeue"
       type="omrelp"
       windowSize="1000"
       action.resumeRetryCount="-1"
       target="remoteserver"
       port="20514"
       queue.filename="somequeue-spool"
       queue.saveonshutdown="on"
       queue.type="LinkedList")
  }
}
```

I upgraded the rsyslog version to 8.10 but i still have the same problem.
Thanks,

**Comment by Smana** [comment]
With the help of David Lang and Brian Knox in the mailing list the issue is now solved.
There is a thread in the mailing list named "Disk queue to flush after restart", the guy had the same problem.
In the debug logs he noticed that some parameters didn't have a default value
```
#### With extra config
5471.816297291:main thread    : name:  'logtoserver'
5471.816299847:main thread    : type:  'omrelp'
5471.816302082:main thread    : action.writeallmarkmessages: (unset)
5471.816304296:main thread    : action.execonlyeverynthtime: (unset)
5471.816306387:main thread    : action.execonlyeverynthtimetimeout: (unset)
5471.816308512:main thread    : action.execonlyonceeveryinterval: (unset)
5471.816310603:main thread    : action.execonlywhenpreviousissuspended:
(unset)
5471.816312746:main thread    : action.repeatedmsgcontainsoriginalmsg:
(unset)
5471.816314841:main thread    : action.resumeretrycount: -1
5471.816317096:main thread    : action.reportsuspension: 1
5471.816319261:main thread    : action.reportsuspensioncontinuation: 1
5471.816321360:main thread    : action.resumeinterval: 10
5471.816323485:main thread    : action.copymsg: (unset)
```
So the workaround is to set a value for each parameter as follows : 
```
ruleset(name="forward"){
    if $programname startswith 'CDN.' then {
        action( type="omrelp"
               name="myapp"
               target="remote-srv"
               port="20514"
               queue.size="5000"
               queue.type="LinkedList"
               queue.spoolDirectory="/var/spool/rsyslog"
               queue.filename="myapp-spool"
               queue.lowwatermark="2000"
               queue.highwatermark="3500"
               queue.discardmark="5000"
               queue.maxfilesize="1g"
               queue.saveonshutdown="on"
               action.ResumeInterval="10"
               action.ResumeRetryCount="-1"
               action.reportSuspension="on"
               action.reportSuspensionContinuation="on"
        )
    }
}
```
Hope it will help

**Comment by rgerhards** [comment]
solved at least in 8.16.0
